analyzer_test.go

```diff
@@ -1,13 +1,649 @@
+					h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
+
+					_, err := analyzer.Analyze()
+					h.AssertNil(t, err)
+
+					h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack", "launch-false.toml"))
+					h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack", "launch-false.sha"))
+				})
+
+				it("does not restore build=true, cache=false layer metadata", func() {
+					h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
+
+					_, err := analyzer.Analyze()
+					h.AssertNil(t, err)
+
+					h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack", "launch-build.sha"))
+				})
+
+				when("subset of buildpacks are detected", func() {
+					it.Before(func() {
+						analyzer.Buildpacks = []buildpack.GroupBuildpack{{ID: "no.cache.buildpack", API: api.Buildpack.Latest().String()}}
+					})
+
+					it("restores layers for detected buildpacks", func() {
+						h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
+
+						_, err := analyzer.Analyze()
+						h.AssertNil(t, err)
+
+						path := filepath.Join(layersDir, "no.cache.buildpack", "some-layer.toml")
+						got := h.MustReadFile(t, path)
+						want := "[metadata]\n  some-layer-key = \"some-layer-value\""
+
+						h.AssertStringContains(t, string(got), want)
+					})
+
+					it("does not restore layers for undetected buildpacks", func() {
+						h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
+
+						_, err := analyzer.Analyze()
+						h.AssertNil(t, err)
+
+						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "metadata.buildpack"))
+					})
+				})
+
+				it("returns the analyzed metadata", func() {
+					md, err := analyzer.Analyze()
+					h.AssertNil(t, err)
+
+					h.AssertEq(t, md.Image.Reference, "s0m3D1g3sT")
+					h.AssertEq(t, md.Metadata, expectedAppMetadata)
+				})
+
+				it("restores each store metadata", func() {
+					h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore store metadata")
+
+					_, err := analyzer.Analyze()
+					h.AssertNil(t, err)
+					for _, data := range []struct{ name, want string }{
+						// store.toml files.
+						{"metadata.buildpack/store.toml", "[metadata]\n  [metadata.metadata-buildpack-store-data]\n    store-key = \"store-val\""},
+						{"no.cache.buildpack/store.toml", "[metadata]\n  [metadata.no-cache-buildpack-store-data]\n    store-key = \"store-val\""},
+					} {
+						got := h.MustReadFile(t, filepath.Join(layersDir, data.name))
+						h.AssertStringContains(t, string(got), data.want)
+					}
+				})
+
+				when("cache exists", func() {
+					it.Before(func() {
```

Comment:
I think you can put the `h.SkipIf` here instead of all of the nested `it` and `when`.